gradle-plugin/plugin/src/main/kotlin/com/emergetools/android/gradle/instrumentation/reaper/ReaperClassLoadTransform.kt

```diff
@@ -83,24 +102,37 @@ class ReaperClassLoadClassVisitor(
     return mv?.let {
       val sig = "$className.$name$descriptor"
       logger.info("Processing method: $sig")
-      ReaperClassLoadMethodVisitor(api, mv, className, name)
+      ReaperClassLoadMethodVisitor(api, mv, className, name, logger, writer)
     }
   }
 }
 
+fun longToBase64(hash: Long): String {
+  val buf = ByteArray(8)
+  for (i in 0..7) {
+    buf[i] = ((hash shr i * 8) and 0xFFL).toByte()
+  }
+  val hashAsBase64 = Base64.getEncoder().encode(buf).toString(Charsets.UTF_8)
+  return hashAsBase64
+}
+
 class ReaperClassLoadMethodVisitor(
   api: Int,
   methodVisitor: MethodVisitor,
   private val className: String,
-  // private val writer: PrintWriter,
   private val name: String?,
+  private val logger: Logger,
+  private val writer: PrintWriter,
 ) : MethodVisitor(api, methodVisitor) {
   override fun visitCode() {
     super.visitCode()
     if (name == "<clinit>" || name == "<init>") {
       val signature = "L" + className.replace(".", "/") + ";"
       val hashedSignature = topLong(toSha256(signature))
 
+      writer.write("${className}\t${hashedSignature}\t${longToBase64(hashedSignature)}\t${signature}\t${name}\n")
```

Comment:
Agh, i realize now that while my example is simliar it is also quite different because my sample is using Gradle’s transforms while this is using AGP’s instrumentation transforms.
If there’s a way to reduce the number of writes, i think that would be better.
This might also be premature optimization (optimisation for you :P ) but can we make it write once per class instead of once per method? If there’s a way to make it write just once at the end of the tranform, that would be even better.